docs/advisors/checks/mongodb-inconsistent-indexes-across-shards.md

```diff
@@ -0,0 +1,35 @@
+# MongoDB Sharding - Inconsistent Indexes Across Sharded Collections
+
+## Description
+This check warns if there are inconsistent indexes across shards for sharded collections. Missing or inconsistent indexes across the shards can have a negative impact on performance.
+
+## Resolution
+
+A sharded collection has an inconsistent index if the collection does not have the exact same indexes (including the index options) on each shard that contains chunks for the collection. Although inconsistent indexes should not occur during normal operations, inconsistent indexes can occur in some situations. Examples of those situations can include the following:
+
+- When a user is creating an index with a unique key constraint and one shard contains a chunk with duplicate documents. In such cases, the create index operation may succeed on the shards without duplicates but fail on the shard with duplicates.
+
+- When a user is creating an index across the shards in a rolling manner (i.e. manually building the index individually across the shards one by one). The index could be inconsistent across all shards if the manual process either fails to build the index for an associated shard or incorrectly builds an index with different specifications for one or more shards. Indexes are often created this way when the shards are very large or when there are a high number of shards.
+
+Starting in MongoDB 4.4 (and 4.2.6), the config server primary checks for index inconsistencies across the shards for sharded collections by default.
+Additionally running the serverStatus command on the config server primary will return the field shardedIndexConsistency to report on the number of sharded collections with index inconsistencies.
+
+If shardedIndexConsistency reports any index inconsistencies, you can identify the missing index or any inconsistent properties across the collection’s shards by running a specific  aggregation pipeline that returns results on inconsistencies. That pipeline script can be found in the “Manage Indexes” section here:  [MongoDB Documentation](https://www.mongodb.com/docs/manual/tutorial/manage-indexes/).
+
+**To resolve the inconsistency where an index is missing from the collection on a particular shard(s), you can either -** 
+
+- Perform a rolling index build for the collection on the affected shard(s) to make sure indexes are built consistently across all involved shards.
+
+(OR)
+
+- Issue an index build db.collection.createIndex() from a mongos instance. The operation only builds the collection's index on the shard(s) missing the index.
+
+**To resolve where the index properties differ across the shards,**
```

Comment:
```suggestion
**The index properties differ across the shards,**
```

Reply:
@nastena1606 We need to keep a like this only, since there are variants of index inconsistency and each has it's own resolution.
Also "##Resolution" is mentioned at the top of the documentation.